common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/FreeformDate.scala

```diff
@@ -0,0 +1,103 @@
+package uk.ac.wellcome.models.parse
+
+import java.time.LocalDate
+
+import uk.ac.wellcome.models.work.internal.InstantRange
+
+/**
+  *  Models freeform text representing some date or period
+  *
+  *  @param date The date or date range
+  *  @param inferred Whether the date is known exactly or inferred
+  */
+case class FreeformDate(date: Either[FuzzyDate, DateRange],
+                        inferred: Boolean = false) {
+
+  def instantRange: Option[InstantRange] =
+    (date match {
+      case Left(date)       => date.instantRange
+      case Right(dateRange) => dateRange.instantRange
+    }) map { _ withInferred inferred }
+
+  def withInferred(inferred: Boolean): FreeformDate =
+    FreeformDate(date, inferred)
+}
+
+/**
+  *  A exact or ambigous date
+  */
+sealed trait FuzzyDate extends DateHelpers {
+
+  def instantRange: Option[InstantRange] =
```

Comment:
I would extract this to a companion object to make the code clearer, also having the effect that the data types have less logic.

Reply:
Yep I agree, cheers